projects/graph/src/graph.py

```diff
@@ -104,3 +105,28 @@ def bfs(self, start, destination):
 
         else:
             print('Those vertices are not in this graph')
+
+    def dfs(self, start, destination):
+        if start in self.vertices and destination in self.vertices:
+            visited = set()
+            stack = [[start]]
+
+            while len(stack):
+                path = stack.pop()
+                vertex = path[-1]
+
+                for neighbor in self.vertices[vertex]:
+                    if neighbor not in visited:
+                        visited.add(neighbor)
```

Comment:
There is a couple of redundancies you could eliminate here by removing the for loop, this sort of thing can just be solved with a while loop